src/Microsoft.ML.Api/DataViewConstructionUtils.cs

```diff
@@ -197,6 +203,18 @@ private Delegate CreateStringToTextGetter(int index)
                         });
                 }
 
+                private Delegate CreateBooleanToDvBoolGetter(int index)
+                {
+                    var peek = DataView._peeks[index] as Peek<TRow, bool>;
+                    Ch.AssertValue(peek);
+                    bool buf = false;
+                    return (ValueGetter<DvBool>)((ref DvBool dst) =>
+                    {
+                        peek(GetCurrentRowObject(), Position, ref buf);
+                        dst = buf ? DvBool.True : DvBool.False;
```

Comment:
>dst = buf ? DvBool.True : DvBool.False; [](start = 24, length = 39)

The shorter `dst = (DvBool)buf;` also works.

Reply:
Given that `DvBool` is a bool with a missing value, might be nice to also handle the `bool?` since it's basically the same concept (just inefficient).

---
In reply to: [189183653](https://github.com/dotnet/machinelearning/pull/183#discussion_r189183653) [](ancestors = 189183653)

Reply:
I added support for bool? but I don't think we have support for any nullable type right now, should I do same for R4, R8, DT?

---
In reply to: [189184119](https://github.com/dotnet/machinelearning/pull/183#discussion_r189184119) [](ancestors = 189184119,189183653)

Reply:
Good question Ivan.... I honestly don't know. Maybe? I'd like to leave it as an open question. If it comes up, it comes up.

---
In reply to: [189339751](https://github.com/dotnet/machinelearning/pull/183#discussion_r189339751) [](ancestors = 189339751,189184119,189183653)